presto-spi/src/main/java/com/facebook/presto/spi/ColumnMetadata.java

```diff
@@ -13,18 +13,22 @@
  */
 package com.facebook.presto.spi;
 
+import com.facebook.presto.spi.statistics.ColumnStatistics;
 import com.facebook.presto.spi.type.Type;
 
 import java.util.Objects;
 
 import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
 
 public class ColumnMetadata
 {
     private final String name;
     private final Type type;
     private final String comment;
     private final boolean hidden;
+    // todo should columnStatistics be part of toString/equals/hashCode
+    private final ColumnStatistics columnStatistics;
```

Comment:
Second thoughts on this one.
Actually I think this field should be removed from here.
And instead `Map<ColumnHandle,ColumnStatistics>` should be added to `ConnectorTableLayout`.
This way we will be able to support different statistics for different set of table partitions in use.